Source/missiles.cpp

```diff
@@ -1263,51 +1243,28 @@ void SetMissDir(int mi, int dir)
 
 void LoadMissileGFX(BYTE mi)
 {
-	MisFileData *mfd = &MissileSpriteData[mi];
-	if (mfd->mAnimData[0] != nullptr)
-		return;
-
-	char pszName[256];
-	if ((mfd->mFlags & MFLAG_ALLOW_SPECIAL) != 0) {
-		sprintf(pszName, "Missiles\\%s.CL2", mfd->mName);
-		byte *file = LoadFileInMem(pszName).release();
-		for (int i = 0; i < mfd->mAnimFAmt; i++)
-			mfd->mAnimData[i] = CelGetFrame(file, i);
-	} else if (mfd->mAnimFAmt == 1) {
-		sprintf(pszName, "Missiles\\%s.CL2", mfd->mName);
-		mfd->mAnimData[0] = LoadFileInMem(pszName).release();
-	} else {
-		for (unsigned i = 0; i < mfd->mAnimFAmt; i++) {
-			sprintf(pszName, "Missiles\\%s%u.CL2", mfd->mName, i + 1);
-			mfd->mAnimData[i] = LoadFileInMem(pszName).release();
-		}
-	}
+	MissileSpriteData[mi].LoadGFX();
 }
 
 void InitMissileGFX()
 {
-	for (int mi = 0; MissileSpriteData[mi].mAnimFAmt != 0; mi++) {
+	for (int mi = 0; MissileSpriteData[mi].animFAmt != 0; mi++) {
 		if (!gbIsHellfire && mi > MFILE_SCBSEXPD)
 			break;
-		if ((MissileSpriteData[mi].mFlags & MFLAG_HIDDEN) == 0)
-			LoadMissileGFX(mi);
+		LoadMissileGFX(mi);

```

Comment:
dont think there is a need to keep this, just call MissileSpriteData[mi].LoadGFX();